### PR TITLE
Unitary thread review

### PR DESCRIPTION
**Pull request by Paul-Licameli**
I reviewed the recent addition of a worker thread.  I correct some errors.  I don't know if more changes we in store for this.  Take selected commits that don't conflict with plans.

### CODE REVIEW

File: src/ProjectFileIO.cpp [hunk 1]
```diff
@@ -332,6 +332,9 @@ void ProjectFileIO::CheckpointThread()
          // Reset
          mCheckpointCurrentPages = 0;
       }
+      else
```
Comment (lllucius) [inline src/ProjectFileIO.cpp:335]: Why?  What is sqlite going to do with a null pointer?  I could see the point if it returned a valid pointer and a non-zero return code.

Comment (lllucius) [reply src/ProjectFileIO.cpp:335]: Indeed, the first thing it does is check for nullpointer then then returns:

static int sqlite3Close(sqlite3 *db, int forceZombie){
  if( !db ){
    /* EVIDENCE-OF: R-63257-11740 Calling sqlite3_close() or
    ** sqlite3_close_v2() with a NULL pointer argument is a harmless no-op. */
    return SQLITE_OK;
  }

Comment (Paul-Licameli) [reply src/ProjectFileIO.cpp:335]: But... we don't know at that point that db is null.  We know that sqlite3_open returned a failure error code.

This page https://www.sqlite.org/c3ref/open.html says:

"Whether or not an error occurs when it is opened, resources associated with the database connection handle should be released by passing it to sqlite3_close() when it is no longer required."

So, a bad rc and a null pointer -- means nothing happens in that call, but there might be a bad rc and a non-null pointer -- which means a memory leak might happen without the call.

**Comment by lllucius** [comment]
Write you are.  Sorry about that.  I went through so many iterations of that thing and I was confusing it with a previous version.  I finally realize that SQLite didn't care if I processed every single hook call.  All he cares about is the last one.  So rather than a queue of requests, I would up simplifying it to a simple trigger.

And apparently I left a lot of leavin's behind.  Thanks for catching it.

**Pull request merged**